packages/javascript/bh-shared-ui/src/components/UpsertOIDCProviderForm.tsx

```diff
@@ -0,0 +1,143 @@
+// Copyright 2024 Specter Ops, Inc.
+//
+// Licensed under the Apache License, Version 2.0
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+import { Button } from '@bloodhoundenterprise/doodleui';
+import { Alert, DialogContent, DialogActions, Grid, TextField } from '@mui/material';
+import { FC } from 'react';
+import { Controller, useForm } from 'react-hook-form';
+import { OIDCProviderInfo, SSOProvider, UpsertOIDCProviderRequest } from 'js-client-library';
+
+const UpsertOIDCProviderDialog: FC<{
+    error?: string;
+    oldSSOProvider?: SSOProvider;
+    onClose: () => void;
+    onSubmit: (data: UpsertOIDCProviderRequest) => void;
+}> = ({ error, oldSSOProvider, onClose, onSubmit }) => {
+    const defaultValues = {
+        name: oldSSOProvider?.name ?? '',
+        client_id: (oldSSOProvider?.details as OIDCProviderInfo)?.client_id ?? '',
+        issuer: (oldSSOProvider?.details as OIDCProviderInfo)?.issuer ?? '',
+    };
+
+    const {
+        control,
+        handleSubmit,
+        reset,
+        formState: { errors },
+    } = useForm<UpsertOIDCProviderRequest>({ defaultValues });
+
+    const handleClose = () => {
+        onClose();
+        reset();
+    };
+
+    return (
+        <form autoComplete='off' onSubmit={handleSubmit(onSubmit)}>
+            <DialogContent>
+                <Grid container spacing={2}>
+                    <Grid item xs={12}>
+                        <Controller
+                            control={control}
+                            name='name'
+                            rules={{
+                                required: !oldSSOProvider && 'OIDC Provider Name is required',
```

Comment:
I see that all the required fields have this `!oldSSOProvider &&` check, does that mean that these fields are not required when updating a given provider record?

Reply:
They are not technically required no, the API will dynamically update only what's changed, that's correct. I suppose I could leave it required 🤷 wouldn't hurt